cmd/soroban-rpc/internal/methods/get_network.go

```diff
@@ -17,20 +19,23 @@ type GetNetworkResponse struct {
 }
 
 // NewGetNetworkHandler returns a json rpc handler to for the getNetwork method
-func NewGetNetworkHandler(daemon interfaces.Daemon, networkPassphrase, friendbotURL string) jrpc2.Handler {
-	coreClient := daemon.CoreClient()
+func NewGetNetworkHandler(
+	logger *log.Entry,
+	networkPassphrase string,
+	friendbotURL string,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+) jrpc2.Handler {
 	return NewHandler(func(ctx context.Context, request GetNetworkRequest) (GetNetworkResponse, error) {
-		info, err := coreClient.Info(ctx)
+		protocolVersion, err := getProtocolVersion(ctx, ledgerEntryReader, ledgerReader)
 		if err != nil {
-			return GetNetworkResponse{}, (&jrpc2.Error{
-				Code:    jrpc2.InternalError,
-				Message: err.Error(),
-			})
+			logger.WithError(err).Info("error occurred while fetching protocol version")
```

Comment:
Shouldn't it be an `Error` log?
```suggestion
			logger.WithError(err).Error("failed to fetch protocol version")
```